app/code/Magento/Catalog/Model/ResourceModel/Category.php

```diff
@@ -723,6 +725,37 @@ public function getCategories($parent, $recursionLevel = 0, $sorted = false, $as
         return $nodes;
     }
 
+    /**
+     * Retrieve categories
+     *
+     * @param integer $parent
+     * @param integer $recursionLevel
+     * @param boolean|string $sorted
+     * @param boolean $asCollection
+     * @param boolean $toLoad
+     * @param boolean $onlyActive
+     * @param boolean $onlyIncludeInMenu
+     * @return \Magento\Framework\Data\Tree\Node\Collection|\Magento\Catalog\Model\ResourceModel\Category\Collection
+     */
+    public function getCategoriesCollection(
```

Comment:
It would be better to introduce the new class with this logic and reuse it in the client code that needs it.

Reply:
Done.